### PR TITLE
Shell updates

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by CameronNemo — APPROVED** [review]

File: pkgs/shells/fish/oh-my-fish/default.nix [hunk 1]
```diff
@@ -2,26 +2,24 @@
 , stdenv
 , fetchFromGitHub
 , fish
-, bash
 , runtimeShell
 , writeShellScript
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
```
Comment (viraptor) [inline pkgs/shells/fish/oh-my-fish/default.nix:9]: If the function version of mkDerivation here used for some specific reason? I don't see it used anywhere else in nixpkgs, so curious about the change.

**Comment by Mindavi** [comment]
I miss the reason to change this from a recursive set to the function used. It doesn't look as nice and I cannot see why it's needed specifically. Also I don't think it's a very common pattern.

Please amend the commit message with a reasoning or revert the change back to using a recursive set.

File: pkgs/shells/fish/oh-my-fish/default.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/shells/fish/oh-my-fish/default.nix:9]: Personal taste. 
I believe `rec` should be deprecated (https://nix.dev/anti-patterns/language#rec-expression), and now that this style of recursive attributes is supported, I am using it instead.
It is more clear than the `let ... in` that the article linked above suggests.

**Comment by AndersonTorres** [comment]
> I miss the reason to change this from a recursive set to the function used.

Explained above.

> It doesn't look as nice and I cannot see why it's needed specifically. Also I don't think it's a very common pattern.

I believe it is as nice as it gets, and also it is not common because its support was added recently.

> Please amend the commit message with a reasoning or revert the change back to using a recursive set.

https://github.com/NixOS/nixpkgs/pull/119942

**Comment by Mindavi** [comment]
Aha! Thanks for the explanation, I saw this in another PR and I totally missed that. Disregard what I said then 😄

**Pull request merged**